add extension 2015-08-01 to templates tags so it passes the ett validator

alexrothenberg opened the pull request:
The ETT Validator https://ttpedge.sitenv.org/ttp/#/validators/ccdar2 does not pass the b2 criteria without these "Extension=2015-08-01" duplicate template tags.  Why? ¯\\_(ツ)_/¯ but let's add them anyway

alexrothenberg committed:
upgrade ruby version to 2.7.1

alexrothenberg committed:
add extension 2015-08-01 to several templates

alexrothenberg commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(2) User doesn't need to add a Plan of Treatment to generate valid CCDA](https://trello.com/c/UGWXQT3n/346-2-user-doesnt-need-to-add-a-plan-of-treatment-to-generate-valid-ccda)

alexrothenberg commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(2) Missing extension causing ETT Errors in Problem section](https://trello.com/c/HXYTlDg8/354-2-missing-extension-causing-ett-errors-in-problem-section)

zelaznik commented on lib/ccd/templates/problem_concern_act_condition_template.rb:
```diff
@@ -10,10 +10,9 @@ def self.included(base)
       # SHALL contain exactly one [1..1] @root="2.16.840.1.113883.10.20.22.4.3" (CONF:16773).
-      constraint 'template_id.root', {:cardinality=>"1..1", :value=>"2.16.840.1.113883.10.20.22.4.3"}
+      constraint 'template_id', {:cardinality=>"1..1", :value=>{:root => "2.16.840.1.113883.10.20.22.4.3"}}
+      constraint 'template_id', {:cardinality=>"1..1", :value=>{:root => "2.16.840.1.113883.10.20.22.4.3", :extension => "2015-08-01"}}
```
It's been a while since I've looked at this library.  If we use the same constraint twice, does one overwrite the other?

alexrothenberg replied:
It generates 2 `templateId` tags which is what's in the official ccds  https://github.com/onc-healthit/2015-certification-ccda-testdata/blob/master/Receiver%20SUT%20Test%20Data/170.315_b2_CIRI_Amb/170.315_b2_ciri__r21_sample1_ccd_v11.xml#L25-L30

```xml
<!-- US Realm Header ID-->
<templateId root="2.16.840.1.113883.10.20.22.1.1" extension="2015-08-01"/>
<templateId root="2.16.840.1.113883.10.20.22.1.1"/>
<!-- CCD template ID-->
<templateId root="2.16.840.1.113883.10.20.22.1.2" extension="2015-08-01"/>
<templateId root="2.16.840.1.113883.10.20.22.1.2"/>
```

alexrothenberg committed:
add 2015-08-01 extension to ccd template

The PI Edge Testing Tool is very particular that the extension be there
https://github.com/onc-healthit/content-validator-api/blob/13a96b59ed7defb7e701332c1608668b064aa39c/src/main/java/org/sitenv/contentvalidator/model/CCDARefModel.java#L73-L75

otherwise we hit this error https://github.com/onc-healthit/content-validator-api/blob/13a96b59ed7defb7e701332c1608668b064aa39c/src/main/java/org/sitenv/contentvalidator/model/CCDARefModel.java#L236

alexrothenberg committed:
version bump 0.2.2

alexrothenberg commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(2) Generated CCDA fails ETT](https://trello.com/c/8feDIPPW/353-2-generated-ccda-fails-ett)

NBallaney commented on lib/ccd/templates/problem_concern_act_condition_template.rb:
```diff
@@ -10,10 +10,9 @@ def self.included(base)
       constraint 'mood_code', {:cardinality=>"1..1", :value=>"EVN"}
       
       # SHALL contain exactly one [1..1] templateId (CONF:16772) such that it
```
Do we want to update this comment and/or the one below it? Same thought for other places we've made this change.

alexrothenberg replied:
What do you think we should change it to?

NBallaney replied:
It feels incomplete because it ends with 'such that it' but.. I looked it up a little and I guess it's okay to leave it as is 👍

NBallaney commented on lib/ccd/templates/problem_concern_act_condition_template.rb:
```diff
@@ -10,10 +10,9 @@ def self.included(base)
       # SHALL contain exactly one [1..1] @root="2.16.840.1.113883.10.20.22.4.3" (CONF:16773).
-      constraint 'template_id.root', {:cardinality=>"1..1", :value=>"2.16.840.1.113883.10.20.22.4.3"}
+      constraint 'template_id', {:cardinality=>"1..1", :value=>{:root => "2.16.840.1.113883.10.20.22.4.3"}}
+      constraint 'template_id', {:cardinality=>"1..1", :value=>{:root => "2.16.840.1.113883.10.20.22.4.3", :extension => "2015-08-01"}}
```
~~I'm guessing it's okay to hard-code the `extension` along with the `template_id`?~~ Also, I'm guessing that they refer to a certain version of a CCDA?

NBallaney replied:
What are your thoughts on making these `ids` constants and storing them in a file dedicated to such constants?

alexrothenberg replied:
I'm scared of making too many changes in this gem because although we "own it now" I don't think we really want to & don't want to invest any more than necessary in it.  So I'd prefer not to start extracting constants if that's ok with you

NBallaney replied:
Cool!

NBallaney commented on ruby-cda.gemspec:
```diff
@@ -4,7 +4,7 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 Gem::Specification.new do |s|
   s.name        = 'ruby-cda'
-  s.version     = '0.2.1'
+  s.version     = '0.2.2'
```
What are your thoughts on creating a change-log?

alexrothenberg replied:
👍

alexrothenberg committed:
add extension 2015-08-01 to the allergies template

alexrothenberg committed:
add us realm template id to the continuity of care document

alexrothenberg committed:
add a changelog

NBallaney reviewed (approved):
🥇
